### PR TITLE
CBL-751 URLEndpointListener Unit Tests

### PR DESCRIPTION
**Pull request by Sandychuang8**
CBL-751 URLEndpointListener Unit Tests

### CODE REVIEW

**Comment by Sandychuang8** [comment]
EE is https://github.com/couchbaselabs/couchbase-lite-net-ee/tree/PR-1232

After reverting the changes in WebSocketWrapper class to 24db4279d9b7a37350007f75e54905152edc22fc and use LiteCore commit 5fd61feaf9280dca61e44aeb8a5d64cc0f66aded [5fd61fe] Tests pass on all platforms.

Not sure why AppVeyor build failed.

File: src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs [hunk 1]
```diff
@@ -43,29 +44,48 @@ public sealed class BasicAuthenticator : Authenticator
...
+        /// [Obsolete("This property is deprecated, please use byte[] PasswordData instead.")]
+        [NotNull]
         public string Password { get; }
 
         /// <summary>
         /// Gets the password that this object holds
         /// </summary>
         [NotNull]
-        public string Username { get; }
+        public byte[] PasswordData { get; }
```
Comment (borrrden) [inline src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs:60]: This is usually the job of the `SecureString` class

Comment (Sandychuang8) [reply src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs:60]: SecureSring works with android and ios too?

File: src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs [hunk 2]
```diff
@@ -638,10 +637,17 @@ private void SetupAuth()
...
-                var password = auth.Password;
-                if (username != null && password != null) {
-                    _logic.Credential = new NetworkCredential(username, password);
+                if (auth.Type == AuthType.HttpBasic) {
+                    var username = auth.Username;
+
+                    // TODO string Password will be deprecated and replaced with byte array password
+                    var password = auth.PasswordData != null ?
+                         Encoding.Unicode.GetString(auth.PasswordData)
+                         : auth.Password;
```
Comment (borrrden) [inline src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs:646]: Use the `SecureString` constructor of `NetworkCredential` for the non-string password

File: src/Couchbase.Lite.Tests.Shared/TLSIdentityTest.cs [hunk 3]
```diff
@@ -0,0 +1,331 @@
+﻿//
+//  TLSIdentityTest.cs
```
Comment (borrrden) [inline src/Couchbase.Lite.Tests.Shared/TLSIdentityTest.cs:2]: Does this file really need to be committed in this PR if it's all just comments?

File: src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs (hunk 1 above)
Comment (borrrden) [reply src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs]: I assume so.  I don't see any reason why it wouldn't.  Do you have information suggesting that it doesn't?

Comment (Sandychuang8) [reply src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs]: Checked, SecureString Class works on all platforms.

File: src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs [hunk 4]
```diff
@@ -77,10 +98,15 @@ internal override void Authenticate([NotNull]ReplicatorOptionsDictionary options
             var authDict = new AuthOptionsDictionary
             {
                 Username = Username,
-                Password = Password,
                 Type = AuthType.HttpBasic
             };
 
+            // TODO string Password will be deprecated and replaced with byte array password
+            if (String.IsNullOrEmpty(Password))
+                authDict.Password = new NetworkCredential(string.Empty, PasswordSecureString).Password;
```
Comment (borrrden) [inline src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs:106]: What is the goal of this?  This defeats the purpose of using `SecureString` by changing it back to a string.  Is it not possible to serialize a secure string?

File: src/Couchbase.Lite.Shared/Sync/AuthOptionsDictionary.cs [hunk 5]
```diff
@@ -61,6 +66,9 @@ internal sealed class AuthOptionsDictionary : OptionsDictionary
 
         #region Properties
 
+        /// <summary>
+        /// [DEPRECATED] Gets the password that this object holds
+        /// </summary>
```
Comment (borrrden) [inline src/Couchbase.Lite.Shared/Sync/AuthOptionsDictionary.cs:71]: Duplicate summary tag

File: src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs (hunk 4 above)
Comment (Sandychuang8) [reply src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs:106]: I am getting `System.ArgumentException : Cannot encode System.Security.SecureString to Fleece!`

**Pull request merged**